File: 1619-path-crossing/path-crossing.py
class Solution:
    def isPathCrossing(self, a: str) -> bool:
        x, y = 0, 0

        coordinate_set = set()
        coordinate_set.add((0, 0))

        for v in a:
            if v == 'N':
                y += 1
            elif v == 'W':
                x -= 1
            elif v == 'E':
                x += 1
            else:
                y -= 1

            if (x, y) in coordinate_set:
                return True

            coordinate_set.add((x, y))

        return False

File: 1619-path-crossing/path-crossing.cpp
class Solution {
public:
    bool isPathCrossing(string a) {
        int x=0,y=0;

        set<pair<int,int>>set;
        set.insert({0,0});

        for(auto v:a){
            if(v=='N') y++;
            else if(v=='W') x--;
            else if(v=='E') x++;
            else y--;

            if(set.find({x,y})!=set.end())
            return true;

            set.insert({x,y});
        }

        return false;
    }
};